firedrake/mesh.py

```diff
@@ -1517,77 +1554,82 @@ class VertexOnlyMeshTopology(AbstractMeshTopology):
+            immersed within a `PETSc.DM` stored in the ``parentmesh``.
+        parentmesh : PETSc.DM
+            Mesh within which the vertex-only mesh topology is immersed.
+        name : str
+            Name of the mesh topology.
+        reorder : bool
+            Whether to reorder the mesh entities.
+        input_ordering_swarm : PETSc.DMSwarm
+            The swarm from which the input-ordering vertex-only mesh is constructed.
+        perm_is : PETSc.IS
+            `PETSc.IS` that is used as ``_dm_renumbering``; only
+            makes sense if we know the exact parallel distribution of ``plex``
+            at the time of mesh topology construction like when we load mesh
+            along with its distribution. If given, ``reorder`` param will be ignored.
+        distribution_name : str
+            Name of the parallel distribution; if `None`, automatically generated.
+        permutation_name : str
+            Name of the entity permutation (reordering); if `None`, automatically generated.
 
-        # Set up the comms the same as the parent mesh
-        self.user_comm = parentmesh.comm
-        self._comm = internal_comm(parentmesh._comm)
-        if MPI.Comm.Compare(swarm.comm.tompi4py(), self._comm) not in {MPI.CONGRUENT, MPI.IDENT}:
+        """
+        if MPI.Comm.Compare(parentmesh.comm, swarm.comm.tompi4py()) not in {MPI.CONGRUENT, MPI.IDENT}:
             ValueError("Parent mesh communicator and swarm communicator are not congruent")
+        self._distribution_parameters = {"partition": False,
+                                         "partitioner_type": None,
+                                         "overlap_type": (DistributedMeshOverlapType.NONE, 0)}
+        self.input_ordering_swarm = input_ordering_swarm
+        super().__init__(swarm, name, reorder, None, perm_is, distribution_name, permutation_name, parentmesh.comm)
+        self._parent_mesh = parentmesh
 
-        # A cache of shared function space data on this mesh
-        self._shared_data_cache = defaultdict(dict)
-
-        # Cell subsets for integration over subregions
-        self._subsets = {}
-
-        tdim = 0
+    def _distribute(self):
+        pass
 
-        cell = ufl.Cell("vertex")
-        self._ufl_mesh = ufl.Mesh(finat.ufl.VectorElement("DG", cell, 0, dim=cell.topological_dimension()))
+    def _add_overlap(self):
+        pass
 
-        # Mark OP2 entities and derive the resulting Swarm numbering
-        with PETSc.Log.Event("Mesh: numbering"):
-            if use_cell_dm_marking:
-                dmcommon.mark_entity_classes_using_cell_dm(self.topology_dm)
-            else:
-                dmcommon.mark_entity_classes(self.topology_dm)
+    def _mark_entity_classes(self):
+        if self.input_ordering_swarm:
+            assert isinstance(self._parent_mesh, MeshTopology)
+            dmcommon.mark_entity_classes_using_cell_dm(self.topology_dm)
+        else:
+            dmcommon.mark_entity_classes(self.topology_dm)
```

Comment:
```suggestion
            # Have an input-ordering vertex-only mesh. These should mark 
            # all entities as pyop2 core, which mark_entity_classes will do.
            assert isinstance(self._parent_mesh, VertexOnlyMeshTopology)
            dmcommon.mark_entity_classes(self.topology_dm)
```
I think here. It makes it clear what's going on to make it clear what's going on